### ISSUE TITLE
In Windows 10 x64 working with timeline slow

### ISSUE THREAD

**Issue by jocamane** [OpenShot/openshot-qt#905]
**System Details:**
* Operating System / Distro: Windows 10 X64
* OpenShot Version: 2.4.0

Is impossible work is this condition.
I think the problem is also in memory manager.
The pc stay in not responding state and wait 1 minute, when I try to do anything with the clips

Thanks

**Comment by Derpito** [comment]
I agree with you, I'm currently (trying) to finish a video but it just continuously says "not responding" and I've literally been editing for over 9 hours, when in windows movie maker or some other online editing software I can finish in less than that. It sucks, I thought it would have been great, I mean it's still easy to use but thats just concerning.

**Comment by DarkYudeX** [comment]
Same here, the software is currently impossible to work with

**Comment by DylanC** [comment]
This is not a place for discussion. We need issues reported with exact details on how to reproduce. There are plenty of other issues with details for us to investigate on Windows 10. Thanks for your understanding.

**Comment by jocamane** [comment]
For my part, rest assured that I have solved the problem and I will not bother again.
Solution: Install other software that does not have bugs of these and that listen to what the users have to say.

Thanks

**Comment by DylanC** [comment]
@jocamane - We do listen but we need proper details for bug reports.  A good bug report contains the information needed to reproduce and fix problems. 

**Bad bug report**
This program is super slow it no work for anything I do.

**Good bug report**
When I used the snip feature to cut a video clip and then moved it to another track the program crashed. The video format was .mp4 and the audio was .mp3. I have screenshots and a video to show the problem to you.  Also my log files are included to help you solve the problem. Thank you.

**Comment by jocamane** [comment]
Please don’t send me more emails.

I'm not your customer anymore

thanks

**Comment by evanworley** [comment]
@DylanC your attitude in this issue and many others is dismissive. I think the general message that openshot is incredibly slow in windows 10 when performing any timeline operations is clear enough. If you tried using it, and doing ANYTHING with the timeline, you'd experience the issues. So please stop shutting everyone down claiming it's not clear. The performance is so bad that you do not need a specific incantation to reproduce.

I've been trying to find a good open source video editor for a few weeks now, and sadly they all disappoint in one way or another. Openshot is the closest to what I need, but as many others have mentioned, performance in windows is very bad.

**Comment by DylanC** [comment]
@evanworley 
You should have a look at the way users behave first before passing judgement. You would think someone like myself would be tired of seeing a negative bug report without any real details to help the developers. Its frustrating and a waste of everyones time.

I'm not shutting down anyone who actually provides details, only people who keep reporting bugs without any specific details to tackle the issue.

Issues seem to be very system specific, so just because a bunch of people have a certain issue doesn't mean we all do. OpenShot has many happy users as well as unhappy too.

I hope you can understand, I'm trying to be balanced and fair. However, I need to make sure bug reports are of a certain quality. Otherwise they are useless to the devs and will not help development or resolution of the problem at all.

**Comment by evanworley** [comment]
@DylanC that's fair, I have not looked through all the open issues. I guess all the people making unhelpful issues can be seen simply as votes, but perhaps closed as duplicates of some other issue. I'm interested in seeing openshot succeed, so can you point me to instructions for what logs or other perf data you'd like me to capture when I am experiencing bad performance?

The use case I have is pretty simple, outlined below
1) Create a new project
2) Import 3-5 existing clips (~30s each), in mp4 format (and put them on a single video channel)
3) Trim each of the clips, typically from both sides
4) Line the clips up back to back
5) Export to MP4

When doing #3 in particular, there are very long delays and often playback freezes. So it ends up taking me 15-20 minutes to trim the clips. Often when I trim one side (say the end) the project is in some bad state and it ends up trimming the beginning. Because of this I now close and re-open the project after every trim. This is the only way that I've found to be stable.

#5 often produces videos with black screens. I am not sure this is related to anything above, but I have noticed that if I do not trim the clips at all, this does not happen.

That's a bit of a long story above, so let me know what data I can collect to help show you what is happening.

**Comment by kkjensen** [comment]
@DylanC @evanworley I can confirm and reproduce what Evan is explaining.   Things seemed to move along half decently until I started snipping.   At that point Windows begins complaining "not responding" and delays in OS responding to commands become very long.   The snips do happen but only after a significant delay.    I've used OS on ubuntu quite a bit and have not experienced these problems so I think it's unique to the windows copy.   

I'm currently using 2.4.1 64 bit edition and bumped the memory caching to 2000MB.  All other settings are at default since installation.

**Comment by nielsswinkels** [comment]
Dylan I understand the need for clear bugreports, but I also agree with the others that this issue arises no matter what you do. Today is the second time I give openshot a try (after the first try ending up in extreme not-responding mess, making me install DaVinci Resolve and then give up all together.)

What I've done now is:
- Import and put on tracks: two mp4 video files (28mb and 464mb) and one wav (36mb).
- Rotated the two video tracks (180 and 90 degrees).
- Made one video 1/4 size
- Enabled "show waveform" on all tracks, even though it only seems to show something on the videos
- Now I'm trying to align the tracks so they are synced. But already here when I try to drag a track the program stops responding for a few seconds, making it rather horrible, and causing me to start googling how to stop OpenShot from being this slow. (ending up here)

This is running on a 64 bit windows 10, quite new MSI gaming laptop with i7 processor and strong graphics card (but Openshot doesn't seem to make use of that?) with power cable plugged in.

What more info do you need? This program seems so well made that I don't want to give up on it, but the unresponsiveness makes it unusable :(

**Comment by Sergix** [comment]
Having the same issues as everyone else on my Windows x64. It seems that OpenShot isn't using my dedicated GPU either... doing the exact same thing as the other people said; just loading in a normal project. Scrubbing, video loading, and anything with the timeline is incredibly slow.

**Comment by HemoglobinSlayer** [comment]
Good day all,
I am having the same problem. As soon as I start snipping clips and adding more than 3 to 4 videos the program becomes unusable. With unusable I mean any action you take on the program takes longer than a minute. 
Think my system specs should be more than efficient enough:

I7 - 7820 CPU @ 2.90 GHz
32gb 2400MHz SODIMM RAM
GTX1070 GPU
512gb SSD
Windows 10 x64

Hope there is a fix. Really like the software.

**Comment by DylanC** [comment]
Possible dup of #902

**Comment by astraldata** [comment]
@DylanC -- This issue is highly-related to snipping a clip and "keeping both sides" of that clip. This happens to me when I do that (i.e. lots of freezing, making the editor impossible to use).

This is the only commonality I've had after reading all the "duplicate" issues, outside of this causing a lot of freezing and being on x64 Windows 10. 

To reproduce this issue of freezing (or as one user put it:  the period of "sleeping"), simply import a video file into a multi-track timeline (mine was .flv format and ~14 mins on some decent hardware), then drag the trackhead to somewhere on the clip, right-click on the video in the timeline and split -> "Keep both sides" and then start dragging the thing around and moving stuff around with it. It could help to add a "title" on a separate track (as this was my exact setup), but it may be unrelated. However, it should start to freeze after modifying the length of (or resizing the beginning/end) of the two clips and trying to overlay them for a fade transition, etc. The only animation I had on mine was for that "title" portion on a separate track. Also, apparently, the length of the clip doesn't have to be long though -- I think the most important thing is to split the clip directly on the track (via Right-click menu -> split) and tell it to "keep both sides" -- this seems to be the only commonality besides Windows 10 -- thus I'd bet money that it's reproducible every time.

The above steps should reproduce the issue everyone is having exactly.

**Comment by n-geoffrey** [comment]
I have the same problem as reported. Windows 10, and after more than 3 object I add (videos, title, or sound track), openshot start becoming unresponsive, very slow. 
Even worst after cutting, but not only. Even without cutting the videos, it starts already to be very slow. 
I have no idea how to help for this, I just find it very sad as this seems to be a very good software...
Maybe we shall switch to linux ? In linux is there the same problem ? 
Kubera

**Comment by peanutbutterandcrackers** [comment]
@n-geoffrey - Switching to Linux is always a good idea. 😆 And, no, there aren't as many problems on Linux with OpenShot as there are with Windows. As a matter of fact, I have successfully created quite a lot of videos on Linux with OpenShot. I hope you have a great time on Linux. Welcome! :)

**Comment by peanutbutterandcrackers** [comment]
@Sergix - Good sir, unfortunately, OpenShot does not have OpenGL capabilities yet. We hope to have that in the future. :)

**Comment by peanutbutterandcrackers** [comment]
The thing about OpenShot is that it is the labour of love of a single man - Mr. Jonathan Thomas. And he has to squirrel time away to work on this amazing software. And because he uses a Linux machine, primarily, windows related issues can not get much attention.
What we need here, ladies and gentlemen, is unity. New volunteer programmers. More financial backers. More people who are willing to support OpenShot through thick and thin.

Also, @DylanC is our only active contributor/issue maintainer at the moment. And he works very hard at this. It is only natural for him to, at times, be frustrated with the issues reported here. Let's not point fingers, ladies and gentlemen, but lets work together for this amazing free software. Let's make sure that at least our posterity will one day make their films using this amazing free/libre software.

"It is not perfect, but it's imperfections are easier to live with than a perfected restrictiveness"... at least on Linux. :)

**Comment by Sergix** [comment]
@peanutbutterandcrackers Ah, ok. Thank you for the clarification!

**Comment by DylanC** [comment]
@peanutbutterandcrackers - Yes, unfortunately sometimes people are not so nice and it can get me a bit grumpy at times. :) Lets all work together and get some progress going!

**Comment by peanutbutterandcrackers** [comment]
@Sergix - You're most welcome.

@DylanC - Yeah! :)

**Comment by coltfreeman** [comment]
I'm in the same boat, win 10 64 bit system, EVERY task I try and perform is met with a significant period of waiting making it unusable. This is a dedicated edit/gameing laptop and I'm a power user attempting to come from a OSX based system to windows. 

Slow Items:
Dragging and dropping clips
Playback in split clip window
Deleting tracks

Watching it in resource monitor it appears to only be allocating around 400mb of RAM and using only about 20% of my CPU. This appears to me having worked on machines that have low resources to be a bottleneck issue. It simply isn't calling enough from my system.

**Comment by nielsswinkels** [comment]
I made an attempt to try to use openshot in a virtual ubuntu on my windows machine, to see if it would work better that way. But then I got lost in the mess of trying to open my project there, with all missing files. I tried fixing the paths by opening the project with a texteditor, but it was to hard to figure out. Then I gave up once again, and now I [learned myself Davinci Resolve](https://www.youtube.com/watch?v=h-WhyR-rfUk) instead. That one works. I hate that I'm advertising for a different program here, but as far as I see this is the only solution at the moment for windows users.

**Comment by peanutbutterandcrackers** [comment]
I see. OpenShot has a lot of issues with Windows and mac... On Linux it works pretty fine for me, though. This has to do with the lack of active developers in the project...

Maybe soon, we'll have more developers joining the project, and one day, hopefully, OpenShot will be the new iMovie and Windows Movie Maker - but better...

**Comment by kpande** [comment]
all of this discussion makes me glad I switched from OpenShot to something more sane, like kdenlive.

I opened an issue with all the required info and @DylanC was nowhere to be found in that thread (#902)

**Comment by DylanC** [comment]
@kpande - That's fine, whatever works for you. I'm not paid to provide support, therefore people should not be expecting all tickets to be resolved really fast.  This project has limited resources and solving performance issues among all the other issues will take time.

**Comment by peanutbutterandcrackers** [comment]
@kpande - It's just that this project is not yet as popular and mature as Kdenlive. We do not have as many volunteers helping out. It's just us scratching each other's back... But we've got more and more volunteer programmers joining us. Perhaps, someday, OpenShot will become as stable as it can be. Maybe you'd also like to join us, someday?

I hope you make awesome videos with kdenlive.

**Comment by ellisonch** [comment]
@DylanC Was this fixed?  Or just closed?

**Comment by DylanC** [comment]
@ellisonch - Just closed. Its reported on a much older version but the original reporter is "not our customer" anymore. So he won't be closing it any time soon.

People have reported much better success with the daily builds but it depends what someone would consider it as "fixed" or not.  Just try a daily build would be the advice. :)

**Comment by David263** [comment]
Sorry this report is closed. I'm having the same issue, extreme slowness in OpenShot catching up even to moving the play head (scrubbing) of just 4 or 5 minutes at 25 frames per second. It is a lot slower than just filling the cache with all the frames in between. I also set cache size to 2000 on a 4GB memory. My version is 2.5.1. Is that a "much older version"? Is there a 32-bit version of OpenShot? Should I submit a bug report? I have to finish a 1-hour video by tomorrow. OpenShot was never this slow on a previous computer.

**Comment by DylanC** [comment]
@David263 - I would recommend to do a search on the issues to try find an issue that describes your exact problem. If you can't find it report a new issue. Your version sounds up to date with the latest release. (Note: I'm not really involved with this project as much any longer)
